src/Host/Client/Impl/Encodings/REncodingsMap.cs

```diff
@@ -0,0 +1,1082 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.R.Host.Client.Encodings {
+    public static class REncodingsMap {
+        // https://msdn.microsoft.com/en-us/library/system.text.encoding.getencodings(v=vs.110).aspx
+        // https://www.ibm.com/support/knowledgecenter/SSMKHH_9.0.0/com.ibm.etools.mft.doc/ac00408_.htm
+        public static Encoding GetEncoding(string rEncodingName) {
+            var encSet = _encodingsMap.Keys.FirstOrDefault(x => x.Contains(rEncodingName));
+            if(encSet != null) {
+                try {
+                    return _encodingsMap[encSet].Invoke();
+                } catch(ArgumentException) { }
+            }
+            return Encoding.Default;
+        }
+
+        private static Dictionary<HashSet<string>, Func<Encoding>> _encodingsMap = new Dictionary<HashSet<string>, Func<Encoding>>() {
```

Comment:
All encodings in .NET also have names. This will be a bit cleaner and easier to add/remove things if it just does name-to-name mapping, and then the code that uses it can call `Encoding.GetEncoding` to get the actual thing.